From 9a7560532a119d9b4a498e2517b9c3e5a857ac25 Mon Sep 17 00:00:00 2001
From: Andriy Tkachuk <andriy_tkachuk@xyratex.com>
Date: Mon, 21 Oct 2013 17:25:56 +0300
Subject: [PATCH 02/17] balloc: balloc_find_by_goal_credit() dropped

balloc_find_by_goal() is not used as by now,
so there is no need to account it in BE credit
calculation.
---
 balloc/balloc.c | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/balloc/balloc.c b/balloc/balloc.c
index 05fc709..2ae8204 100644
--- a/balloc/balloc.c
+++ b/balloc/balloc.c
@@ -1359,13 +1359,6 @@ static int balloc_free_db_update(struct m0_balloc *mero,
 	M0_RETURN(rc);
 }
 
-static void
-balloc_find_by_goal_credit(const struct m0_balloc *bal, int nr,
-				 struct m0_be_tx_credit *accum)
-{
-	balloc_db_update_credit(bal, nr, accum);
-}
-
 static int balloc_find_by_goal(struct m0_balloc_allocation_context *bac)
 {
 	m0_bindex_t group = balloc_bn2gn(bac->bac_goal.e_start,
@@ -1707,12 +1700,11 @@ balloc_alloc_credit(const struct m0_ad_balloc *balroom, int nr,
 {
 	const struct m0_balloc	*bal = b2m0(balroom);
 
-	M0_ENTRY("cred=%lux%lu nr=%d",
+	M0_ENTRY("cred=[%lu:%lu] nr=%d",
 		(unsigned long)accum->tc_reg_nr,
 		(unsigned long)accum->tc_reg_size, nr);
-	balloc_find_by_goal_credit(bal, nr, accum);
 	balloc_db_update_credit(bal, nr, accum);
-	M0_LEAVE("cred=%lux%lu",
+	M0_LEAVE("cred=[%lu:%lu]",
 		(unsigned long)accum->tc_reg_nr,
 		(unsigned long)accum->tc_reg_size);
 }
@@ -1895,7 +1887,7 @@ int balloc_allocate_internal(struct m0_balloc *ctx,
 
 	while (req->bar_len &&
 	       !balloc_claim_free_blocks(ctx, req->bar_len)) {
-		req->bar_len = req->bar_len >> 1;
+		req->bar_len >>= 1;
 	}
 	if (req->bar_len == 0) {
 		rc = -ENOSPC;
-- 
1.8.3.2

